Further remove redundant CMake option passing code for those CMake variables that are directly controlled by environment variables but with a different name.

xuhdev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22154 Further remove redundant CMake option passing code for those CMake variables that are directly controlled by environment variables but with a different name.**
* #22153 Remove more build options not needed to be explicitly set in Python build scripts.

This is done by making `additional_options` a dict rather than a set.

Differential Revision: [D15985376](https://our.internmc.facebook.com/intern/diff/D15985376)

xuhdev committed:
Further remove redundant CMake option passing code for those CMake variables that are directly controlled by environment variables but with a different name.

ezyang commented on tools/setup_helpers/cmake.py:
```diff
@@ -239,23 +239,35 @@ def generate(self, version, cmake_python_library, build_python, build_test, my_e
+            '_GLIBCXX_USE_CXX11_ABI': 'GLIBCXX_USE_CXX11_ABI',
+            'MKL_SEQ': 'INTEL_MKL_SEQUENTIAL',
+            'MKL_TBB': 'INTEL_MKL_TBB',
+            'USE_CUDA_STATIC_LINK': 'CAFFE2_STATIC_LINK_CUDA'
```
Sigh. Our build system really is outrageous XD

ezyang commented:
This and co all waiting on CI

xuhdev committed:
Update on "Further remove redundant CMake option passing code for those CMake variables that are directly controlled by environment variables but with a different name."

Further remove redundant CMake option passing code for those CMake variables that are directly controlled by environment variables but with a different name.

gh-metadata: pytorch pytorch 22154 gh/xuhdev/5/head

xuhdev commented:
Updated to resolve a newly developed conflict in master

xuhdev committed:
Update on "Further remove redundant CMake option passing code for those CMake variables that are directly controlled by environment variables but with a different name."

Further remove redundant CMake option passing code for those CMake variables that are directly controlled by environment variables but with a different name.

gh-metadata: pytorch pytorch 22154 gh/xuhdev/5/head
